components/calendar/calendar-header.component.ts

```diff
@@ -25,55 +27,60 @@ import { NzSelectModule, NzSelectSizeType } from 'ng-zorro-antd/select';
   selector: 'nz-calendar-header',
   exportAs: 'nzCalendarHeader',
   template: `
-    <div class="ant-picker-calendar-header">
-      <nz-select
-        class="ant-picker-calendar-year-select"
-        [nzSize]="size"
-        [nzDropdownMatchSelectWidth]="false"
-        [ngModel]="activeYear"
-        (ngModelChange)="updateYear($event)"
-      >
-        @for (year of years; track year.value) {
-          <nz-option [nzLabel]="year.label" [nzValue]="year.value" />
-        }
-      </nz-select>
-
-      @if (mode === 'month') {
+    @if (nzCustomHeader) {
+      <ng-container [ngTemplateOutlet]="nzCustomHeader"></ng-container>
```

Comment:
Use nzStringTemplateOutlet instead of ngTemplateOutlet, which allows nzCustomHeader to accept both strings and TemplateRef types.